packages/cyberstorm/src/components/Pagination.tsx

```diff
@@ -0,0 +1,140 @@
+import { faArrowLeft, faArrowRight } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import React, { Dispatch, SetStateAction } from "react";
+import { range } from "../utils";
+import styles from "./componentStyles/Pagination.module.css";
+import { PaginationButton } from "./PaginationButton";
+
+export interface PaginationProps {
+  currentPage: number;
+  onPageChange: Dispatch<SetStateAction<number>>;
+  totalCount: number;
+  pageSize: number;
+  disabled?: boolean;
+  siblingCount?: number;
+}
+
+const DOTS = "…";
+
+const getButtonLabels = (
+  totalPageCount: number,
+  currentPage: number,
+  siblingCount = 1
+) => {
+  // maxButtons is determined as siblingCount + firstPage + lastPage + currentPage + 2*DOTS
+  const maxButtons = siblingCount + 5;
+
+  // Show a button for each page if there's just a few of them.
+  if (maxButtons >= totalPageCount) {
+    return range(1, totalPageCount);
+  }
+
+  const prevPageNumber = Math.max(currentPage - siblingCount, 1);
+  const nextPageNumber = Math.min(currentPage + siblingCount, totalPageCount);
+
+  const shouldShowLeftDots = prevPageNumber > 2;
+  const shouldShowRightDots = nextPageNumber < totalPageCount - 2;
+
+  const firstPageIndex = 1;
+  const lastPageIndex = totalPageCount;
+
+  if (shouldShowLeftDots && shouldShowRightDots) {
+    const middleRange = range(prevPageNumber, nextPageNumber);
+    return [firstPageIndex, DOTS, ...middleRange, DOTS, lastPageIndex];
+  } else if (shouldShowRightDots) {
+    const leftItemCount = 3 + 2 * siblingCount;
+    const leftRange = range(1, leftItemCount);
+    return [...leftRange, DOTS, totalPageCount];
+  } else if (shouldShowLeftDots) {
+    const rightItemCount = 3 + 2 * siblingCount;
+    const rightRange = range(
+      totalPageCount - rightItemCount + 1,
+      totalPageCount
+    );
+    return [firstPageIndex, DOTS, ...rightRange];
+  } else {
+    return [];
+  }
+};
+
+export const Pagination: React.FC<PaginationProps> = (props) => {
+  const {
+    currentPage,
+    onPageChange,
+    totalCount,
+    pageSize,
+    disabled,
+    siblingCount = 1,
+  } = props;
+  const totalPageCount = Math.ceil(totalCount / (pageSize > 0 ? pageSize : 1));
+  const paginationRange = getButtonLabels(
+    totalPageCount,
+    currentPage,
+    siblingCount
+  );
+
+  function decreaseCurrentPage(currentPage: number) {
+    const y = currentPage - 1;
```

Comment:
Nitpick: while the variable name has little meaning when the scope where it's used is whole two lines, I think renaming it to `newPage` oslt might improve readability? This is not a hard-and-fast rule btw, the issue should be judged case by case.

Reply:
Got your point and changed it to match newPage. Will try my luck on other cases to see where is that line when it's ok 😉